common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java

```diff
@@ -234,8 +234,22 @@ public K waitForDeployment(String name) {
 
     @Override
     public K waitForPod(String name) {
+        // wait when all pods are ready
         return waitFor("pod", name,
-            actualObj -> "Running".equals(actualObj.get("status").get("phase").asText()));
+            actualObj -> {
+                try {
+                    JsonNode containerStatuses = actualObj.get("status").get("containerStatuses");
+                    if (containerStatuses.isArray()) {
```

Comment:
`if (containerStatuses != null && containerStatuses.isArray()) {` then you don't need the `catch (NPE)`

Reply:
The code looks match better without try/catch for NPE